### PR TITLE
[cairo] bump dependencies + support conan v2

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **cairo**

Update the autotools build to the new toolchains and bump dependencies.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by planetmarshall** [comment]
Requires `libgettext` and `fontconfig` (#13845) to be updated otherwise various Autotools checks fail on Macos due to requiring `layout` to be implemented in those recipes.

File: recipes/cairo/all/conanfile.py [hunk 1]
```diff
@@ -72,167 +77,205 @@ def config_options(self):
...
+        self.settings.rm_safe("compiler.libcxx")
 
     def validate(self):
-        if microsoft.is_msvc(self):
+        if is_msvc(self):
             if self.settings.build_type not in ["Debug", "Release"]:
                 raise ConanInvalidConfiguration("MSVC build supports only Debug or Release build type")
-            if self.options.get_safe("with_glib") and self.options["glib"].shared \
-                    and microsoft.is_msvc_static_runtime(self):
+            if self.options["glib"].shared and is_msvc_static_runtime(self):
```
Comment (uilianries) [inline recipes/cairo/all/conanfile.py:88]:
```suggestion
            if self.dependencies["glib"].options.shared and is_msvc_static_runtime(self):
```

File: recipes/cairo/all/conanfile.py [hunk 2]
```diff
@@ -72,167 +77,205 @@ def config_options(self):
...
-            self.copy(pattern="cairo.h", dst=inc, src=src)
-            self.copy(pattern="cairo-deprecated.h", dst=inc, src=src)
-            self.copy(pattern="cairo-win32.h", dst=inc, src=src)
-            self.copy(pattern="cairo-script.h", dst=inc, src=src)
-            self.copy(pattern="cairo-ft.h", dst=inc, src=src)
-            self.copy(pattern="cairo-ps.h", dst=inc, src=src)
-            self.copy(pattern="cairo-pdf.h", dst=inc, src=src)
-            self.copy(pattern="cairo-svg.h", dst=inc, src=src)
-            self.copy(pattern="cairo-gobject.h", dst=inc, src=cairo_gobject)
+        copy(self, pattern="LICENSE", dst="licenses", src=self.source_folder)
```
Comment (uilianries) [inline recipes/cairo/all/conanfile.py:241]:
```suggestion
        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```

**Comment by uilianries** [comment]
Both PRs #14520 and #14521 should fix those versions conflicts.

**Comment by uilianries** [comment]
@planetmarshall that error is odd. I listed to be investigated internally. Please, go ahead reverting it, because we can re-apply those changes easily in the future.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by toge — APPROVED** [review]

**Pull request merged**